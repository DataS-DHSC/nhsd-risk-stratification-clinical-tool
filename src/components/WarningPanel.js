import React from 'react';
import PropTypes from 'prop-types';

function overrideWarningMessage(originalMessage) {
  let overriddenWarning = {};
  switch (originalMessage) {
    case 'The provided BMI is below the calculators range. The minimum valid BMI will be used instead.': // Adds apostrophe
      overriddenWarning =
        "The provided BMI is below the calculator's range. The minimum valid BMI will be used instead.";
      break;
    case 'The provided BMI is above the calculators range. The maximum valid BMI will be used instead.': // Adds apostrophe
      overriddenWarning =
        "The provided BMI is above the calculator's range. The maximum valid BMI will be used instead.";
      break;
    default:
      overriddenWarning = originalMessage;
  }
  return overriddenWarning;
}

const WarningPanel = ({ warnings }) => (
  <>
    {warnings.length > 0 ? (
      <div id="warning-panel" className="nhsuk-inset-text">
        <p>
          We used defaults for the values you did not enter, or were outside our
          range.
        </p>
        <p>We did this for:</p>
        <ul>
          {warnings.map((warning) => (
            <li
              key={warning.coding.code}
              id={`warning-${warning.coding.system}-${warning.coding.code}`}
            >
              {overrideWarningMessage(warning.message)}
            </li>
          ))}
        </ul>
      </div>
    ) : (
      ''
    )}
  </>
);

WarningPanel.propTypes = {
  warnings: PropTypes.arrayOf(
    PropTypes.shape({
      severity: PropTypes.string,
      message: PropTypes.string,
    })
  ),
};

WarningPanel.defaultProps = {
  warnings: [],
};

export default WarningPanel;
