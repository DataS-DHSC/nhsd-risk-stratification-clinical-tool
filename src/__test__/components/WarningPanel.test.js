import React from 'react';
import { render } from '@testing-library/react';
import WarningPanel from '../../components/WarningPanel';

const setup = (warnings) => {
  const props = {
    warnings,
  };

  return render(<WarningPanel {...props} />);
};

describe('WarningPanel', () => {
  it('empty should render blank fragment ', () => {
    const container = setup([]);
    expect(container.asFragment()).toMatchSnapshot();
  });

  it('warnings should render', () => {
    const container = setup([
      {
        coding: {
          code: '206',
          system: 'http://qcovid.org',
        },
        message: 'No BMI was provided. A BMI of 25 will be used.',
        severity: 'warning',
      },
      {
        coding: {
          code: '301',
          system: 'http://qcovid.org',
        },
        message:
          'No postcode was provided. A Townsend score of 0 will be used.',
        severity: 'warning',
      },
    ]);

    expect(container.asFragment()).toMatchSnapshot();
  });
});
