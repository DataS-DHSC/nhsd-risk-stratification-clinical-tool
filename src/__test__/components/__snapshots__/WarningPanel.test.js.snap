// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WarningPanel empty should render blank fragment  1`] = `<DocumentFragment />`;

exports[`WarningPanel warnings should render 1`] = `
<DocumentFragment>
  <div
    class="nhsuk-inset-text"
    id="warning-panel"
  >
    <p>
      We used defaults for the values you did not enter, or were outside our range.
    </p>
    <p>
      We did this for:
    </p>
    <ul>
      <li
        id="warning-http://qcovid.org-206"
      >
        No BMI was provided. A BMI of 25 will be used.
      </li>
      <li
        id="warning-http://qcovid.org-301"
      >
        No postcode was provided. A Townsend score of 0 will be used.
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
